File: android/src/main/java/org/ligi/gobandroid_hd/ui/fragments/GobandroidGameAwareFragment.java
package org.ligi.gobandroid_hd.ui.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.ligi.gobandroid_hd.App;
import org.ligi.gobandroid_hd.events.GameChangedEvent;
import org.ligi.gobandroid_hd.logic.GoGame;

public abstract class GobandroidGameAwareFragment extends GobandroidFragment {

    protected GoGame game;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        App.component().inject(this);
        game = gameProvider.get();
        EventBus.getDefault().register(this);
        return createView(inflater, container, savedInstanceState);
    }

    public abstract View createView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState);

    @Override
    public void onDestroyView() {
        EventBus.getDefault().unregister(this);
        super.onDestroyView();
    }

    @Subscribe
    public void onGoGameChanged(GameChangedEvent gameChangedEvent) {

    }

}

File: android/src/main/java/org/ligi/gobandroid_hd/ui/fragments/GobandroidGameAwareFragment.kt
package org.ligi.gobandroid_hd.ui.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode
import org.ligi.gobandroid_hd.events.GameChangedEvent
import org.ligi.gobandroid_hd.logic.GoGame

abstract class GobandroidGameAwareFragment : GobandroidFragment() {

    protected lateinit var game: GoGame

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        game = gameProvider.get()
        EventBus.getDefault().register(this)
        return createView(inflater!!, container, savedInstanceState)
    }

    abstract fun createView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View

    override fun onDestroyView() {
        EventBus.getDefault().unregister(this)
        super.onDestroyView()
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    open fun onGoGameChanged(gameChangedEvent: GameChangedEvent?) {

    }

}
